File: Binary Search.py
from typing import List


class Solution:
    def search(self, nums: List[int], target: int) -> int:
        low=0
        high=len(nums)-1
        mid=0

        while low<=high:
            mid = (high+low)//2

            if nums[mid]>x:
                high=mid-1
            elif nums[mid]<x:
                low=mid+1
            else:
                return mid
        return -1

File: binarysearch.c
 int search(int* nums, int numsSize, int target){
    int l=0,h=numsSize-1,m;
    while (l<=h){
        m=(h+l)/2;
        if (nums[m]==target){
            return m;
        }else if(nums[m]<target){
            l=m+1;
        }else{
            h=m-1;
        }
    }
    return -1;
}
int main(void){
    return 0;
}